### PR TITLE
feature/historic-messages

### PR DESCRIPTION
**Pull request by decanus**
This is the experimentation PR for the new Historic Messages, everything relevant will be added to the spec PR.

Spec PR: https://github.com/vacp2p/specs/pull/147

### CODE REVIEW

**Comment by decanus** [comment]
@oskarth this is the PR.

**Comment by oskarth** [comment]
@decanus can you please fill out PR description with what it is doing and link to relevant spec PR? Makes it easier to understand the scope of change and to see context / have discussion...

File: waku/protocol/v2/waku_historic_messages.nim [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+type
+  Cursor* = ref object
+    before*: seq[byte]
+    after*: seq[byte]
+
+  Query* = ref object
+    origin*: int32
+    to*: int32
+    bloom*: seq[byte]
```
Comment (oskarth) [inline waku/protocol/v2/waku_historic_messages.nim:17]: This seems specific to v1

File: waku/protocol/v2/waku_historic_messages.nim [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import 
+  libp2p/protocols/protocol,
+  libp2p/protobuf/minprotobuf,
+  stew/results
+
+const
+  HistoricMessagesCodec = "/waku/history/1.0.0"
```
Comment (oskarth) [inline waku/protocol/v2/waku_historic_messages.nim:7]: For a spike OK, but we need to make sure these work fine together. See https://github.com/vacp2p/specs/issues/87

Probably worth adding -alpha or so here as well and make sure this is in spec

File: waku/protocol/v2/waku_historic_messages.nim (hunk 1 above)
Comment (decanus) [reply waku/protocol/v2/waku_historic_messages.nim:17]: Hmm, have we decided not to use bloom filters? if so @oskarth, how do we reduce meta data leakage to mailservers?

**Comment by oskarth** [comment]
Ping @decanus

**Comment by decanus** [comment]
@oskarth this is still an active working branch where I am currently playing around.

I currently have a few blocking issues, mainly what is the message format we receive from nodes?

**Comment by oskarth** [comment]
Look at the issues and PRs I created and ping you at. The message payload is already specified in the spec.

**Pull request closed without merging**